src/libs/conduit/conduit_node_iterator.cpp

```diff
@@ -135,6 +137,24 @@ NodeIterator::index() const
 }
 
 //---------------------------------------------------------------------------//
+std::string
+NodeIterator::id() const
```

Comment:
@cyrush: I added this new `conduit::NodeIterator::id` function in order to simplify the process of tagging the children of a node that can either be an object or a list.  Does this concept seem reasonable (i.e. a string identifier that works for both objects and lists) and, if so, does the name `id` seem adequately descriptive?  If not, please let me know and I'll modify this pull request.